compiler/src/dmd/funcsem.d

```diff
@@ -573,666 +559,691 @@ void funcDeclarationSemantic(Scope* sc, FuncDeclaration funcdecl)
 
     if (ClassDeclaration cd = parent.isClassDeclaration())
     {
-        parent = cd = objc.getParent(funcdecl, cd);
+        switch (classFuncSemantic(cd, funcdecl, parent, sc, f))
+        {
+            case 0: break;
```

Comment:
What are these magic numbers doing here???

An enum would be great.

Reply:
An enum for a private function that is called from only one place is a bit overkill.
 I've documented the function itself as to what it returns and why.